packages/cojson/src/tests/coValueCore.isCompletelyDownloaded.test.ts

```diff
@@ -0,0 +1,491 @@
+    childGroup.extend(parentGroup1);
+    childGroup.extend(parentGroup2);
+
+    const map = childGroup.createMap();
+    map.set("test", "value");
+
+    await map.core.waitForSync();
+
+    // Disconnect the admin node to sync the content manually
+    admin.disconnect();
+
+    // Make both parent groups large enough to require multiple messages to be synced
+    for (let i = 0; i < 100; i++) {
+      // @ts-expect-error - test property is not part of the group shape
+      parentGroup1.set(`key1_${i}`, "1".repeat(1024));
+      // @ts-expect-error - test property is not part of the group shape
+      parentGroup2.set(`key2_${i}`, "1".repeat(1024));
+    }
+
+    const content1 = parentGroup1.core.verified.newContentSince(undefined);
+    const content2 = parentGroup2.core.verified.newContentSince(undefined);
+    assert(content1);
+    assert(content2);
+    expect(content1.length).toBeGreaterThan(1);
+    expect(content2.length).toBeGreaterThan(1);
+
+    const lastChunk1 = content1.pop();
+    const lastChunk2 = content2.pop();
+    assert(lastChunk1);
+    assert(lastChunk2);
+
+    // Create a new session for bob
+    const bobSession = bob.node;
+
+    // Feed all chunks except the last ones from both parents
+    for (const chunk of content1) {
+      bobSession.syncManager.handleNewContent(chunk, "import");
+    }
+    for (const chunk of content2) {
+      bobSession.syncManager.handleNewContent(chunk, "import");
+    }
+
+    const mapOnBob = await loadCoValueOrFail(bobSession, map.id);
+
+    // The map should not be completely downloaded because both parents are still streaming
+    expect(mapOnBob.core.isCompletelyDownloaded()).toBe(false);
+
+    // Feed the last chunk from first parent
+    bobSession.syncManager.handleNewContent(lastChunk1, "import");
+
+    // Still should not be completely downloaded because second parent is still streaming
+    await new Promise<void>((resolve) => queueMicrotask(resolve));
+    expect(mapOnBob.core.isCompletelyDownloaded()).toBe(false);
+
+    // Feed the last chunk from second parent
+    bobSession.syncManager.handleNewContent(lastChunk2, "import");
+
+    // Now it should be completely downloaded
+    await new Promise<void>((resolve) => queueMicrotask(resolve));
+    expect(mapOnBob.core.isCompletelyDownloaded()).toBe(true);
+  });
+
+  test("should return false when the source branch of the value is streaming", async () => {
+    const group = admin.node.createGroup();
+    const originalMap = group.createMap();
+
+    // Add initial data to original map
+    originalMap.set("key1", "value1");
+
+    await originalMap.core.waitForSync();
+
+    // Disconnect the admin node to sync the content manually
+    admin.disconnect();
+
+    // Make the original map large enough to require multiple messages to be synced
+    for (let i = 0; i < 100; i++) {
+      originalMap.set(`key${i}`, "1".repeat(1024));
+    }
+
+    // Create a branch from the original map
+    const branch = expectMap(
```

Comment:
Is this situation the same if we first branch the map and then add new content to the original map? I.e. do we always need to wait for streaming of the original covalue to complete.

It shouldn't be necessary in this second case, right? If so, don't think it's something we need to optimize right now, but maybe we should have a separate test case for that scenario.